presto-main/src/main/java/com/facebook/presto/server/protocol/Query.java

```diff
@@ -507,14 +509,18 @@ private synchronized void setQueryOutputInfo(QueryExecution.QueryOutputInfo outp
 
     private synchronized URI createNextResultsUri(String scheme, UriInfo uriInfo, long nextToken)
     {
-        return uriInfo.getBaseUriBuilder()
+        UriBuilder uri = uriInfo.getBaseUriBuilder()
                 .scheme(scheme)
                 .replacePath("/v1/statement/executing")
                 .path(queryId.toString())
                 .path(String.valueOf(nextToken))
                 .replaceQuery("")
-                .queryParam("slug", slug)
-                .build();
+                .queryParam("slug", this.slug);
+        Optional<DataSize> targetResultSize = getTargetResultSize(session);
+        if (targetResultSize.isPresent()) {
+            uri = uri.queryParam("targetResultSize", targetResultSize.get());
```

Comment:
How this param is going to be used?

Reply:
It's used here: https://github.com/prestodb/presto/blob/master/presto-main/src/main/java/com/facebook/presto/server/protocol/ExecutingStatementResource.java#L152

This will change the batch size to be streamed back from the coordinator to the client.

This change could have been made differently.  We could have chosen to not add this as a query parameter, and instead use it directly inside the coordinator (without parsing it from the URI).  The intent is to set a default from the coordinator to be sent back to the client.  Because there's already a query parameter for this in `ExecutingStatementResource`, it seemed natural to simply add it to the `nextUri`.